sympy/solvers/solvers.py

```diff
@@ -1014,7 +1072,7 @@ def _sympified_list(w):
         # to be obtained to queries like solve((x - y, y), x); without
         # this mod the return value is []
         ok = False
-        if fi.free_symbols & symset:
+        if fi.has(*symset):
```

Comment:
These are not equivalent wrt bound symbols. We should think of cases where this might not work and consider carefully whether this change is always valid. Also the performance impact should be checked in benchmarks. The important case is large sparse separable systems.

Reply:
> These are not equivalent wrt bound symbols. 

What is a bound symbol?

Reply:
> What is a bound symbol?

`x` in `Integral(x + y, (x, 1, 3))` is bound while `y` is a free symbol. And an expression might have the same symbol appearing as free and bound as in `x + Integral(x, (x, 1, 3))`.

Reply:
Also if symset is large then `has` can be very slow compared to `free_symbols`.